specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json

```diff
@@ -1201,6 +1201,433 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AgFoodPlatform/farmBeats/{farmBeatsResourceName}/solutions/{solutionId}": {
+      "put": {
+        "tags": [
+          "Solutions"
+        ],
+        "description": "Install Or Update Solution.",
+        "operationId": "Solutions_CreateOrUpdate",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FarmBeatsResourceNameParameter"
+          },
+          {
+            "in": "path",
+            "name": "solutionId",
+            "description": "Solution Id of the solution.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z]{3,50}[.][a-zA-Z]{3,100}$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Solution resource request body.",
+            "schema": {
+              "$ref": "#/definitions/SolutionInstallationRequest"
```

Comment:
>SolutionInstallationRequest [](http://example.com/codeflow?start=37&length=27)

Why not same request and response schema?
SolutionInstallationRequest vs Solution?

Reply:
SolutionInstallationRequest is a subset of Solution model. We just wanted to show only the required fields in body.

Reply:
Request and Response models must be the same for PUt